lib/v2/yahoo/news/index.js

```diff
@@ -9,38 +9,46 @@ module.exports = async (ctx) => {
         throw Error('Invalid region');
     }
     const category = ctx.params.category ? ctx.params.category.toLowerCase() : '';
-    const rssUrl = `https://${region ? region + '.' : ''}news.yahoo.com/rss/${category}`;
+    const rssUrl = `https://${region ? `${region}.` : ''}news.yahoo.com/rss/${category}`;
     const feed = await parser.parseURL(rssUrl);
-    const title = feed.title;
-    const link = feed.link;
-    const description = feed.description;
-    const filteredItems = feed.items.filter((item) => !item.link.includes('promotions') && new URL(item.link).hostname.match(/.*\.yahoo\.com$/));
+    const filteredItems = feed.items.filter(
+        (item) => !item.link.includes('promotions') && new URL(item.link).hostname.match(/.*\.yahoo\.com$/)
+    );
     const items = await Promise.all(
-        filteredItems.map(async (item) => {
-            const description = await ctx.cache.tryGet(item.link, async () => {
-                const response = await got({
-                    method: 'get',
-                    url: item.link,
-                });
+        filteredItems.map(
+            async (item) => {
+                try {
+                    return await ctx.cache.tryGet(item.link, async () => {
+                        const response = await got({
+                            method: 'get',
+                            url: item.link,
+                        });
+
+                        const $ = cheerio.load(response.data);
+                        const author = `${$('span.caas-author-byline-collapse').text()} @${$('span.caas-attr-provider').text()}`;
+
+                        $('.caas-content-byline-wrapper, .caas-xray-wrapper, .caas-header, .caas-readmore').remove();
+                        const description = $('.caas-content-wrapper').html();
 
-                const $ = cheerio.load(response.data);
-                $('.caas-content-byline-wrapper, .caas-xray-wrapper, .caas-header, .caas-readmore').remove();
-                return $('.caas-content-wrapper').html();
-            });
-            const single = {
-                title: item.title,
-                description,
-                pubDate: item.pubDate,
-                link: item.link,
-            };
-            return single;
-        })
+                        const single = {
+                            title: item.title,
+                            description,
+                            author,
+                            pubDate: item.pubDate,
+                            link: item.link,
+                        };
+                        return single;
+                    });
+                } catch (e) {
+                    return Promise.resolve('');
+                }
+            })
```

Comment:
[No return await](https://eslint.org/docs/latest/rules/no-return-await)
```suggestion
        filteredItems.map((item) =>
            ctx.cache.tryGet(item.link, async () => {
                const response = await got({
                    method: 'get',
                    url: item.link,
                });

                const $ = cheerio.load(response.data);
                const author = `${$('span.caas-author-byline-collapse').text()} @${$('span.caas-attr-provider').text()}`;

                $('.caas-content-byline-wrapper, .caas-xray-wrapper, .caas-header, .caas-readmore').remove();
                const description = $('.caas-content-wrapper').html();

                const single = {
                    title: item.title,
                    description,
                    author,
                    pubDate: item.pubDate,
                    link: item.link,
                };
                return single;
            })
        )
```

Reply:
我不擅长js，但根据 [no-return-await](https://eslint.org/docs/latest/rules/no-return-await) 
```
// In this example the `await` is necessary to be able to catch errors thrown from `bar()`
async function foo() {
    try {
        return await bar();
    } catch (error) {}
}
```
当然，如果开发团队认为 `ctx.cache.tryGet` 根本没必要在外侧await，我可以改掉

Reply:
This is what you will have in your PR if an error occurs:

<details><summary>Code screenshot</summary>

![image](https://github.com/DIYgod/RSSHub/assets/11386903/747f6db4-7d6a-4786-ab15-f8e1f0a7b140)
</details>

<details><summary>RSS screenshot</summary>

![image](https://github.com/DIYgod/RSSHub/assets/11386903/d36ead03-9e4a-47fa-b2a7-e25a103db787)
</details>

<details><summary>Console log</summary>

![image](https://github.com/DIYgod/RSSHub/assets/11386903/bbcfe52b-f57d-4968-903e-2b418ec26898)
</details>

---

This is what you will have in your PR if you don't return `Promise.resolve('')` in `catch` and  an error occurs:

<details><summary>Code screenshot</summary>

![image](https://github.com/DIYgod/RSSHub/assets/11386903/797513f3-b7ca-4961-abdd-0846ef25738b)
</details>

<details><summary>RSS screenshot</summary>

![image](https://github.com/DIYgod/RSSHub/assets/11386903/6e949f78-ff33-4e14-a9ed-d4d5d02abc7a)
</details>

---

This is what happens in the suggested changes if an error occurs:

<details><summary>Code screenshot</summary>

![image](https://github.com/DIYgod/RSSHub/assets/11386903/ac3b9178-0c7b-48e7-bd17-fbb72c7f1e0b)
</details>

<details><summary>RSS screenshot</summary>

![image](https://github.com/DIYgod/RSSHub/assets/11386903/98a32371-07ae-4b91-b40a-815e6b200420)
</details>

Reply:
谢谢指点，我明白为什么要改了，这就去改